lib/adapters/http.js

```diff
@@ -47,6 +48,56 @@ function dispatchBeforeRedirect(options) {
   }
 }
 
+/**
+ * Provides an HTTP tunnel socket via an HTTP CONNECT request.
+ * The tunnel socket is used for sending HTTPS requests via an HTTP proxy.
```

Comment:
HTTP requests can be made via an HTTP proxy using the CONNECT method as well. Shouldn't this use either a `Socket` or `TlsSocket` depending on the destination protocol?

Reply:
@Nevon exactly, CONNECT is only used if the destination uses TLS. I figured it would not make that much sense for non-TLS destinations because of the extra round trip.